src/components/AppProvider/README.md

```diff
@@ -431,14 +431,59 @@ function AppProviderWithAllThemeKeysExample() {
 Translations are provided in the locales folder. When using Polaris, you are able to import translations from all languages supported by the core Shopify product and consume them through the `i18n` prop.
 
 ```jsx
-// en.json is English. Replace with fr.json for French, etc
```

Comment:
I think referencing a `translations` variable without showing where it comes from is confusing. Let's keep this as an example of the static translations, then segue into dynamic translations - that way we might not need the explicit headings.

How does something like this sound? (Sorry for only coming up with this after you wrote the first draft):


> Translations are provided in the locales folder. When using Polaris, you are able to import translations from all languages supported by the core Shopify product and consume them through the `i18n` prop.
> 
> At its simplest you can pass the json content from a locale file into `AppProvider`. This approach is suitable when a project has only one specific locale.
> 
> ```jsx
> import AppProvider from "@shopify/polaris";
> // en.json is English. Replace with fr.json for French, etc
> import translations from "@shopify/polaris/locales/en.json";
> 
> function App() {
>   return <AppProvider i18n={translations}>{/* App content */}</AppProvider>;
> }
> ```
> 
> Alternatively if your project supports multiple locales you can load them dynamically using [`@shopify/react-i18n`](https://github.com/Shopify/quilt/tree/master/packages/react-i18n#translation). This will ensure you only load the translations you need instead of potentially loading every set of translations even if you are not using them.
> 
> ```jsx
> import AppProvider from "@shopify/polaris";
> // en.json is English. Replace with fr.json for French, etc
> import translations from "@shopify/polaris/locales/en.json";
> import { useI18n } from "@shopify/react-i18n";
> 
> function App() {
>   const [i18n] = useI18n({
>     id: "Polaris",
>     fallback: translations,
>     translations(locale) {
>       return import(
>         /* webpackChunkName: "Polaris-i18n-[request]", webpackMode: "lazy-once" */ `@shopify/polaris/locales/${locale}.json`
>       ).then(dictionary => dictionary && dictionary.default);
>     }
>   });
> 
>   // i18n.translations is an array of translation dictionaries, where the first
>   // dictionary is the desired language, and the second is the fallback.
>   // AppProvider however expects that the first dictionary is the fallback
>   // and the second is the desired language. Thus we need to reverse the array
>   // to ensure the dictionaries are in the order desired by AppProvider
>   return (
>     <AppProvider i18n={i18n.translations.reverse()}>{/* App content */}</AppProvider>
>   );
> }
> ```

---

I've modified the translations function there, but I've not ran it to test as I've not got an environment setup for that. Would you be able to check it for me - it is based up on a simplified version of the code the [react-i18n babel plugin](https://github.com/Shopify/quilt/blob/3f8f237de601f93501f6a2ad0d63bdbe845ea43c/packages/react-i18n/src/babel-plugin.ts#L30-L48) generates - that's where the "webpacMode" comes from. 

I've removed any typescript types as we want our examples to be plain JS. I've also removed the `const translationsLocale = locale && locale !== '' ? locale : 'en';` check as [react-i18n's types](https://github.com/Shopify/quilt/blob/3f8f237de601f93501f6a2ad0d63bdbe845ea43c/packages/react-i18n/src/manager.ts#L9-L11) suggests that a string will always be passed into the translate function. I've also added a comment explaining why the reverse is needed inline, rather than explaining it in the docs (related - we should fix the ordering to match usei18n in the next major version to avoid this fiddling)

Reply:
agree that fixing the order will help - then there will be no need to understand this reverse "gotcha"